themes/default/content/docs/concepts/inputs-outputs/apply.md

```diff
@@ -0,0 +1,822 @@
+
+vpc.vpc_id.apply(lambda id: print('VPC ID:', id))
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+{{< example-program-snippet path="awsx-vpc" language="go" from="1" to="3" >}}
+    "fmt"
+{{< example-program-snippet path="awsx-vpc" language="go" from="4" to="10" >}}
+{{< example-program-snippet path="awsx-vpc" language="go" from="12" to="15" >}}
+
+        vpc.VpcId().ApplyT(func(id string) error {
+            fmt.Printf("VPC ID: %s", id)
+        	return nil
+        })
+
+{{< example-program-snippet path="awsx-vpc" language="go" from="21" to="23" >}}
+```
+
+{{% notes %}}
+**A note on error handling** The function `ApplyT` spawns a Goroutine to await the availability of the implicated dependencies. This function accepts a `T` or `(T, error)` signature; the latter accomodates for error handling. Alternatively, one may use the `ApplyTWithContext` function in which the provided context can be used to reject the output as canceled. Error handling may also be achieved using an `error` `chan`.
+{{% /notes %}}
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+{{< example-program-snippet path="awsx-vpc" language="csharp" from="1" to="6" >}}
+{{< example-program-snippet path="awsx-vpc" language="csharp" from="8" to="8" >}}
+
+    vpc.VpcId.Apply(id => { Console.WriteLine($"VPC ID: {id}"); return id; });
+
+{{< example-program-snippet path="awsx-vpc" language="csharp" from="17" to="17" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+```java
+{{< example-program-snippet path="awsx-vpc" language="java" from="1" to="9" >}}
+{{< example-program-snippet path="awsx-vpc" language="java" from="11" to="11" >}}
+
+            vpc.vpcId().applyValue(i -> {
+                System.out.println("VPC ID: " + i);
+                return null;
+            });
+
+{{< example-program-snippet path="awsx-vpc" language="java" from="17" to="19" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+# YAML does not have the Apply method, but you can access values directly.
+{{< example-program-snippet path="awsx-vpc" language="yaml" from="1" to="4" >}}
+{{< example-program-snippet path="awsx-vpc" language="yaml" from="6" to="7" >}}
+{{< example-program-snippet path="awsx-vpc" language="yaml" from="9" to="9" >}}
+{{< example-program-snippet path="awsx-vpc" language="yaml" from="11" to="11" >}}
+```
+
+{{% /choosable %}}
+
+The above example will wait for the value to be returned from the API and print it to the console as shown below:
+
+```bash
+Updating (pulumi/dev)
+
+     Type                 Name         Status     Info
+     pulumi:pulumi:Stack  aws-iac-dev             1 message
+
+Diagnostics:
+  pulumi:pulumi:Stack (aws-iac-dev):
+    VPC ID: vpc-0f8a025738f2fbf2f
+
+Resources:
+    34 unchanged
+
+Duration: 12s
+```
+
+We can now see the value of the VPC ID property that we couldn't see before when using a regular `print | log` statement.
+
+### Creating new output values
```

Comment:
I really like this use case; I wonder would we consider putting it above the VPC example?

Reply:
@interurban Can you clarify what you mean here? I'm not sure I understand.